### ISSUE TITLE
[boto] [botocore] importing boto/botocore before gevent patching raises exception

### ISSUE THREAD

**Issue by brettlangdon** [DataDog/dd-trace-py#662]
Reported in https://github.com/DataDog/dd-trace-py/issues/506#issuecomment-431692318

This is a similar problem as #632 

To reproduce:

```python
# app.py
import gevent.monkey
gevent.monkey.patch_all()
```

```
$ ddtrace-run python app.py
app.py:2: MonkeyPatchWarning: Monkey-patching ssl after ssl has already been imported may lead to errors, including RecursionError on Python 3.6. It may also silently lead to incorrect behaviour on Python 3.7. Please monkey-patch earlier. See https://github.com/gevent/gevent/issues/1016. Modules that had direct imports (NOT patched): ['botocore.httpsession (/Users/brett.langdon/.env/issue-662/lib/python3.7/site-packages/botocore/httpsession.py)', 'botocore.vendored.requests.packages.urllib3.util.ssl_ (/Users/brett.langdon/.env/issue-662/lib/python3.7/site-packages/botocore/vendored/requests/packages/urllib3/util/ssl_.py)', 'botocore.vendored.requests.packages.urllib3.util (/Users/brett.langdon/.env/issue-662/lib/python3.7/site-packages/botocore/vendored/requests/packages/urllib3/util/__init__.py)', 'urllib3.util (/Users/brett.langdon/.env/issue-662/lib/python3.7/site-packages/urllib3/util/__init__.py)', 'urllib3.util.ssl_ (/Users/brett.langdon/.env/issue-662/lib/python3.7/site-packages/urllib3/util/ssl_.py)'].
  gevent.monkey.patch_all()
```

**Comment by rollokb** [comment]
Hey @brettlangdon! 

If you're looking for a solution to this. I found it's because botocore while being patched gets imported, and imports some of it's own dependencies, including urllib.

The solution is a little bit messy. So with `ddtrace-run gunicorn -k gevent` I found the solution was to disable botocore, and anything else that imports urllib like so:

```
DATADOG_PATCH_MODULES=botocore:false,elasticsearch:false,requests:false ddtrace-run gunicorn -k gevent app
```
Then explicitly patch them again in your app.py

```
import ddtrace
ddtrace.patch(
    requests=True, botocore=True, elasticsearch=True
)
```

Also make sure you're using gevent 1.3.4 or greater to silence the now needless monkeypatch warning.

:)

**Comment by majorgreys** [comment]
We can address this by enabling patching on import for more of our integrations, particularly those that have `urllib3` as a dependency. Once that's completed, this workaround will not be necessary.

**Comment by lior-neuman** [comment]
We're on our first day of evaluating APM; this bug kills the application, so in our case it's a blocker.
Any update on this?

**Comment by 5ay3h** [comment]
Hi DD people,

I'm also experiencing this issue with:
```
gunicorn==20.0.4
gevent==20.6.2
datadog==0.38.0
ddtrace==0.40.0
```
Executing like: 
```
ddtrace-run gunicorn --worker-class=gevent ...
```

So to summarise, this issue is still open, with only [this solution](https://github.com/DataDog/dd-trace-py/issues/662#issuecomment-482080556) (that btw fixes my problem also).

The major problem here is this issue not being referred anywhere in your setup docs: http://pypi.datadoghq.com/trace/docs/async_integrations.html#gevent or https://docs.datadoghq.com/tracing/setup/python/ .

So if it wont be fixed anytime soon (already more than a year 🙃), I think you should document it properly, as it's a blocker for us (all gevent and boto/requests users). 

@brettlangdon 🙏

**Comment by Kyle-Verhoog** [comment]
Hey all, I put a fix up in #1629 if anyone could verify it with their particular configuration that would be great 🙂

We plan to include it in `v0.42` but it can be tried right now in master:

```
pip install git+https://github.com/datadog/dd-trace-py.git@master
```
